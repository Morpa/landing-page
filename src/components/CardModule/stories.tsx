import { Story, Meta } from '@storybook/react/types-6-0'
import CardModule, { CardModuleProps } from '.'

export default {
  title: 'CardModule',
  component: CardModule,
  args: {
    title: 'title default',
    subTitle: 'description default'
  }
} as Meta

export const Default: Story<CardModuleProps> = (args) => (
  <CardModule {...args}>
    <div>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Quasi culpa, odit
      rerum optio impedit quae aspernatur excepturi consequuntur assumenda
      temporibus quos iure similique ullam officia delectus fuga corporis illo
      omnis.
    </div>
  </CardModule>
)
