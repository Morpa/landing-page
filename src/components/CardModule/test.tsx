import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import CardModule from '.'

const props = {
  title: 'React',
  subTitle: 'ReactRockstar'
}

describe('<CardModule />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(
      <CardModule {...props}>lorem ipsun</CardModule>
    )

    expect(
      screen.getByRole('heading', { name: props.title })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', { name: props.subTitle })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
