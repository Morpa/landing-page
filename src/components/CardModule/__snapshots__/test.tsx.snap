// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardModule /> should render the heading 1`] = `
.c0 {
  background-color: #FFFFFF;
  padding: 3.2rem;
  color: #030517;
  font-size: 1.6rem;
  border-radius: 0.4rem;
}

.c1 {
  border-left: 6px solid #F231A5;
  padding-left: 1.6rem;
  margin-bottom: 1.6rem;
}

.c2 {
  font-size: 2.2rem;
  font-weight: bold;
}

.c3 {
  font-size: 1.4rem;
  font-weight: normal;
  color: #7B7373;
}

.c4 p + p {
  margin-top: 2.4rem;
}

<article
  class="c0"
>
  <div
    class="c1"
  >
    <h3
      class="c2"
    >
      React
    </h3>
    <h4
      class="c3"
    >
      ReactRockstar
    </h4>
  </div>
  <div
    class="c4"
  >
    lorem ipsun
  </div>
</article>
`;
